src/senaite/sync/browser/views.py

```diff
@@ -230,13 +236,21 @@ def import_data(self, domain):
                     obj = self.create_object_slug(container, data)
                     # remember the UID -> object UID mapping for the update step
                     uidmap[uid] = api.get_uid(obj)
+                    objmap[uid] = obj
+
+        transaction.commit()
```

Comment:
Please add a comment in code why you need to do a `transaction.commit()` here in between.